src/sinks/datadog/metrics/encoder.rs

```diff
@@ -428,11 +414,18 @@ fn encode_timestamp(timestamp: Option<DateTime<Utc>>) -> i64 {
 fn generate_series_metrics(
     metric: &Metric,
     default_namespace: &Option<Arc<str>>,
+    log_schema: &'static LogSchema,
     last_sent: Option<Instant>,
 ) -> Result<Vec<DatadogSeriesMetric>, EncoderError> {
     let name = get_namespaced_name(metric, default_namespace);
+
+    let mut tags = metric.tags().cloned().unwrap_or_default();
+    let host = tags.remove(log_schema.host_key());
+    let source_type_name = tags.remove("source_type_name");
+    let device = tags.remove("device");
     let ts = encode_timestamp(metric.timestamp());
-    let tags = metric.tags().map(encode_tags);
+    let tags = Some(encode_tags(&tags));
+    // let tags = metric.tags().map(encode_tags);
```

Comment:
Probably should remove this throw-away comment

Reply:
Nice catch, will remove.